File: go-client-sdk/pkg/models/shared/codatdatacontractsdatasetscustomerpagedresponsemodel.go
package shared

// CodatDataContractsDatasetsCustomerPagedResponseModel
// Used to represent what can be returned by an endpoint that supports paging.
// Usable with the [ProducesResponseType] attribute on a controller action.
type CodatDataContractsDatasetsCustomerPagedResponseModel struct {
	Links        *CodatDataContractsDatasetsCustomerPagedResponseLinksModel `json:"_links,omitempty"`
	PageNumber   *int                                                       `json:"pageNumber,omitempty"`
	PageSize     *int                                                       `json:"pageSize,omitempty"`
	Results      []CodatDataContractsDatasetsCustomer                       `json:"results,omitempty"`
	TotalResults *int                                                       `json:"totalResults,omitempty"`
}

File: typescript-client-sdk/src/sdk/models/shared/codatdatacontractsdatasetscustomerpagedresponsemodel.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatDataContractsDatasetsCustomer } from "./codatdatacontractsdatasetscustomer";
import { CodatDataContractsDatasetsCustomerPagedResponseLinksModel } from "./codatdatacontractsdatasetscustomerpagedresponselinksmodel";
import { Expose, Type } from "class-transformer";


// CodatDataContractsDatasetsCustomerPagedResponseModel
/** 
 * Used to represent what can be returned by an endpoint that supports paging.
 * Usable with the [ProducesResponseType] attribute on a controller action.
**/
export class CodatDataContractsDatasetsCustomerPagedResponseModel extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "_links" })
  @Type(() => CodatDataContractsDatasetsCustomerPagedResponseLinksModel)
  links?: CodatDataContractsDatasetsCustomerPagedResponseLinksModel;

  @SpeakeasyMetadata()
  @Expose({ name: "pageNumber" })
  pageNumber?: number;

  @SpeakeasyMetadata()
  @Expose({ name: "pageSize" })
  pageSize?: number;

  @SpeakeasyMetadata({ elemType: CodatDataContractsDatasetsCustomer })
  @Expose({ name: "results" })
  @Type(() => CodatDataContractsDatasetsCustomer)
  results?: CodatDataContractsDatasetsCustomer[];

  @SpeakeasyMetadata()
  @Expose({ name: "totalResults" })
  totalResults?: number;
}